src/main/java/filter/Parser.java

```diff
@@ -107,18 +106,10 @@ private FilterExpression parseExpression(int precedence) {
         return left;
     }
 
-    private FilterExpression parseQuotedKeywords() {
-        FilterExpression result = parseQuotedContent(QualifierType.KEYWORD);
-        consume(TokenType.QUOTE);
-        return result;
-    }
-
-    private FilterExpression parseQuotedContent(QualifierType type) {
-        StringBuilder sb = new StringBuilder();
-        while (!isQuoteToken(lookAhead())) {
-            sb.append(consume().getValue()).append(' ');
-        }
-        return new Qualifier(type, sb.toString().trim());
+    private FilterExpression parseQuotedContent(QualifierType type, Token token) {
+        String keywords = token.getValue();
+        keywords = keywords.substring(1, keywords.length() - 1);
```

Comment:
Instead of this (which doesn't always remove the correct characters), we can use `Utility.stripQuotes` and `trim`.